Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -0,0 +1,137 @@
+
+public class MainActivity extends AppCompatActivity {
+    private Integer totalFrames=0;
+    private Integer safeFrames=10;
+    private final Integer FILE_PICKER_REQUEST_CODE = 10;
+
+    private TextView filePath;
+    private Button filePickerBtn;
+    private Button analyzeBtn;
+    private Intent fileIntent;
+    private MediaMetadataRetriever mediaMetadataRetriever;
+    private Uri fileUri;
+    private InputImage imageHolder;
+    private TextRecognizer recognizer;
+    private ArrayList<String> results = new ArrayList<String>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        filePath = findViewById(R.id.selectedFilePath);
+        filePickerBtn = findViewById(R.id.fileSelectButton);
+        analyzeBtn = findViewById(R.id.analyze);
+
+        recognizer = TextRecognition.getClient();
+        mediaMetadataRetriever = new MediaMetadataRetriever();
+        Bundle extras = getIntent().getExtras();
+
+        try {
+            filePath.setText(extras.getString("file URI"));
+        }catch (Exception e){
+            Log.d("Rokus logs:", "Integration failed, no file URI received from capture session");
+        }
+
+        filePickerBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                fileIntent = new Intent(Intent.ACTION_GET_CONTENT);
+                fileIntent.setType("*/*");
+                startActivityForResult(fileIntent, FILE_PICKER_REQUEST_CODE);
+            }
+        });
+
+        analyzeBtn.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.P)
+            @Override
+            public void onClick(View view) {
+                try {
+                    analyze();
+                }catch (Exception e){
+                    e.printStackTrace();
+                    e.getCause();
+                }
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == FILE_PICKER_REQUEST_CODE && resultCode == RESULT_OK) {
+            String path = data.getData().toString();
+            fileUri = data.getData();
+            filePath.setText(path);
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.P)
+    private void analyze(){
+        fileUri = Uri.parse(filePath.getText().toString());
+        mediaMetadataRetriever.setDataSource(getApplicationContext(), fileUri);
+        totalFrames = Integer.valueOf(mediaMetadataRetriever.extractMetadata(METADATA_KEY_VIDEO_FRAME_COUNT));
+        List<Bitmap> frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames-safeFrames));
+
+        for(int i=0; i<frameList.size();i++) {
```

Comment:
nit: Still seeing some strange formatting here and elsewhere. Can you run the formatter over this and the other files you changed?

Reply:
can you set up either format on save or format on commit?